packages/mobile/src/components/Avatar.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * Essentially the same as @celo/react-components/components/Avatar but with
+ * a fallback to an unknown user icon.  Must be in mobile since the images are
+ * in the mobile package.
+ */
+
+import { Avatar as PlainAvatar } from '@celo/react-components/components/Avatar'
+import * as React from 'react'
+import { withNamespaces, WithNamespaces } from 'react-i18next'
+import { Image, StyleSheet } from 'react-native'
+import { Namespaces } from 'src/i18n'
+import { unknownUserIcon } from 'src/images/Images'
+import { getRecipientThumbnail, Recipient, RecipientKind } from 'src/recipients/recipient'
+
+interface Props {
+  recipient?: Recipient
+  name?: string
+  address?: string
+  e164PhoneNumber?: string
+  defaultCountryCode: string
+  iconSize?: number
+}
+
+type AvatarProps = Props & WithNamespaces
+
+export class Avatar extends React.PureComponent<AvatarProps> {
+  render() {
+    const { t, recipient, address, e164PhoneNumber, defaultCountryCode, iconSize = 40 } = this.props
+
+    let { name } = this.props
+
+    if (!recipient && !name) {
+      // TransferFeedItem does not specify what kind of recipient was used, so
```

Comment:
Rather than make this assumption, can we just have TransferFeedItem specify the recipient kind?